src/main/scala/io/citrine/lolo/learners/RandomForest.scala

```diff
@@ -68,13 +77,16 @@ case class RandomForest(
           numFeatures = numFeatures,
           minLeafInstances = minLeafInstances,
           maxDepth = maxDepth,
-          splitter = RegressionSplitter(randomizePivotLocation)
+          splitter = RegressionSplitter(randomizePivotLocation),
+          rng = rng
```

Comment:
What happens to this random number generator when the the training of the trees in the random forest is parallelized?

Reply:
Paraphrasing slack discussion for posterity: parallel training breaks reproducibility. For now, this can be addressed by disabling parallelism when perfect reproducibility is required. A possible improvement, probably only worth undertaking if customers start to demand reproducibility, is to implement a threaded counter-based PRNG. (AFAICT, no such implementations exist for the JVM.)

Reply:
Why would this be necessary? This approach (essentially splitting the PRNG) should work:
https://github.com/google/jax/blob/master/design_notes/prng.md

Reply:
Maybe relevant: reproducibility in distributed environments is implemented in smlb: https://github.com/CitrineInformatics/smlb/pull/15/files

Reply:
Yep! Implementing a split approach is exactly what I meant. The main advantage of counter-based PRNGs is the ability to split them into an arbitrary number of streams that are both statistically and computationally independent. Linear PRNGs, including `scala.util.Random` and the Mersenne Twister algorithm implemented by `np.random.RandomState`, do not have this property. They are designed only to produce a single stream of statistically independent random numbers, whereas multiple streams initialized with different seeds are not statistically independent from each other. This is why the Jax notes cite the Claessen & Pałka paper, which presents a counter-based PRNG: https://dl.acm.org/doi/10.1145/2578854.2503784

But there is good news! I was being too narrow in my search terms (looking for specific algorithms) and now I've found this: https://docs.oracle.com/javase/8/docs/api/java/util/SplittableRandom.html
It's not immediately clear what algorithm it uses, but at least it passes DieHarder. That's good enough to convince me it's suitable for us, barring unexpected shortcomings in its interface or in its computational overhead. Still, I think that's outside the scope of the purpose of this PR, which is to reduce the noise of our gating tests. We can revisit if we encounter a specific motivating case for reproducibility that clearly also benefits from parallelism.

Reply:
Java's `SplittableRandom` currently uses SplitMix, but it's [possible that this algorithm may change in future Java versions](http://openjdk.java.net/jeps/356), namely to a different algorithm called "LXM" (but I know of no article or paper that describes LXM, as opposed to SplitMix, in more detail).